app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux_autoinstall.erb

```diff
@@ -34,6 +34,9 @@ test_on:
   end
   options << "locale=#{host_param('lang') || 'en_US'}"
   options = options.join(' ')
+  userdata_server = foreman_url('').sub(/.*:\/\//, '').split('/').first
+  userdata_server = @preseed_server if userdata_server.split(':').first == foreman_server_fqdn
```

Comment:
Oh, I think… You need `foreman_request_addr` from `app/services/foreman/foreman_url_renderer.rb` here.

This will resolve to the right name/port combination of Foreman or Proxy with the right port for the template feature.